multinet/util.py

```diff
@@ -4,14 +4,39 @@
 
 from flask import Response
 
-from typing import Sequence, Any, Generator
+from typing import Sequence, Any, Generator, Dict, Set
 
 from . import db
 from .errors import DatabaseNotLive, DecodeFailed
+from .types import EdgeTableProperties
 
 TEST_DATA_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../test/data"))
 
 
+def get_edge_table_properties(workspace: str, edge_table: str) -> EdgeTableProperties:
+    """Return extracted information about an edge table."""
+
+    loaded_workspace = db.db(workspace)
+    edges = list(loaded_workspace.collection(edge_table).all())
+
+    tables_to_keys: Dict[str, Set] = {}
+    from_tables = set()
+    to_tables = set()
+
+    for edge in edges:
+        nodes = (edge["_from"].split("/"), edge["_to"].split("/"))
```

Comment:
The code in this block might be a bit more readable if you try

```suggestion
        from_node, to_node = (edge["_from"].split("/"), edge["_to"].split("/"))
```

Of course you'd have to modify the remainder of the block to follow suit.